validphys2/src/validphys/config.py

```diff
@@ -774,35 +773,59 @@ def produce_loaded_theory_covmat(
         data_input,
         theory_covmat_flag=False,
         use_user_uncertainties=False,
-        use_scalevar_uncertainties=True
+        use_scalevar_uncertainties=True,
     ):
         """
         Loads the theory covmat from the correct file according to how it
         was generated by vp-setupfit. 
         """
         if theory_covmat_flag is False:
-            return 0.
-        #Load correct file according to how the thcovmat was generated by vp-setupfit
+            return 0.0
+        # Load correct file according to how the thcovmat was generated by vp-setupfit
         generic_path = "datacuts_theory_theorycovmatconfig_theory_covmat_custom.csv"
         if use_user_uncertainties is True:
             if use_scalevar_uncertainties is True:
-                generic_path = "datacuts_theory_theorycovmatconfig_total_theory_covmat.csv"
+                generic_path = (
+                    "datacuts_theory_theorycovmatconfig_total_theory_covmat.csv"
+                )
             else:
-                generic_path = "datacuts_theory_theorycovmatconfig_user_covmat.csv"   
-        theorypath = output_path/"tables"/generic_path
+                generic_path = "datacuts_theory_theorycovmatconfig_user_covmat.csv"
+        # check if there are multiple files
+        files = glob.glob(str(output_path / "tables/*theorycovmat*"))
+        paths = [
+            str(
+                output_path
+                / "tables/datacuts_theory_theorycovmatconfig_theory_covmat_custom.csv"
+            ),
+            str(
+                output_path
+                / "tables/datacuts_theory_theorycovmatconfig_total_theory_covmat.csv"
+            ),
+            str(
+                output_path
+                / "tables/datacuts_theory_theorycovmatconfig_user_covmat.csv"
+            ),
+        ]
+        paths.remove(str(output_path / "tables" / generic_path))
+        for f in files:
+            for path in paths:
+                if f == path:
+                    raise ValueError(
+                        "More than one theory_covmat file in folder tables"
+                    )
```

Comment:
@andreab1997 this doesn't seem right?

Reply:
Why not?

Reply:
The error is not as general as I originally thought. It probably does do what you intended, but I tried it with a custom covmat and got this error because I did not set `use_scalevar_uncertainties` to false, in which case this message does not make a lot of sense since there was only one covmat. 

So I guess we just need to update the example here: https://github.com/NNPDF/nnpdf/blob/278310410398b36b95e45246992c8ed07db6d6a6/doc/sphinx/source/tutorials/general_th_covmat.rst? If that's correct, I can update the docs.

P.S. the check can be written in one line for example as
`
if set(paths)&set(files):
    raise ValueError
`

Reply:
Ok yes I agree. I would say the check is checking what I wanted (even if it can be written better) but the error message is misleading.

Reply:
I guess there is no situation in which both `use_scalevar_uncertainties` and `use_user_uncertainties` are true? If so, that should probably be checked by vp.

Reply:
For the moment I would say no but in principle yes (for example one can have a different source of theory uncertainties which are not `user_uncertainties`)

Reply:
Sure, but here `use_scalevar_uncertainties` is just synonym for using a covmat that has been constructed from different theoryID's (regardless of the source of the theory uncertainty) with the instructions in the runcard, while `user_uncertainties` is an external covmat (not necessarily non-scalevar).

So for now I think `use_scalevar_uncertainties` can probably be removed/deprecated as it is just the inverse of `user_uncertainties`?

Reply:
If you agree, I can take care of the changes. Just you know this better than me, so it's good if you can confirm :)

Reply:
Yes I agree. In my mind we could have a third option, like `use_whatever_uncertainties`, that uses a different prescription from the `scalevar`. However it is true that we do not have now so for the time being `use_scalevar_uncertainties` is just not `user_uncertainties`. So yes, I agree :)

Reply:
Let's leave hypothetical future features  for hypothetical future people to work on ;)

Ok, I'll open a PR some point this week and ask for your review.